contrib/segmentio/kafka.go.v0/kafka.go

```diff
@@ -46,13 +48,17 @@ func WrapReader(c *kafka.Reader, opts ...Option) *Reader {
 	if c.Config().Brokers != nil {
 		wrapped.bootstrapServers = strings.Join(c.Config().Brokers, ",")
 	}
+
+	wrapped.groupID = c.Config().GroupID
```

Comment:
@adrien-f Comparing with `Shopify/sarama` contrib, `groupID` comes from [`WithGroupID`](https://github.com/DataDog/dd-trace-go/blob/4904f57e55983c8cbf261f9c7b1561fc04071729/contrib/Shopify/sarama/option.go#L63) instead. I think it'll be better to be coherent with the other contrib, and avoid changing the default behaviour (working without GroupID).

Do you mind adding the option and removing this line?

Reply:
For sure, I'm also looking at the `confluent-kafka-go` integration and they have a WithConfig: 

https://github.com/DataDog/dd-trace-go/blob/4904f57e55983c8cbf261f9c7b1561fc04071729/contrib/confluentinc/confluent-kafka-go/kafka/option.go#L114-L116

Which is called through this:

https://github.com/DataDog/dd-trace-go/blob/4904f57e55983c8cbf261f9c7b1561fc04071729/contrib/confluentinc/confluent-kafka-go/kafka/kafka.go#L43-L44

And I like this approach of enhancing the constructor transparently as users adopting DSM will only need to update and turn on the environment variable instead of having to explicitly pass the groupID. 

This might be too much of a broad change for this PR though, what do you think?

Reply:
@adrien-f If you want, I can contribute the change. It won't take me too much time. I'd rather avoid introducing a subtle breaking change.

Reply:
No worries, I just wanted to be sure :) I've pushed the change!

Reply:
@adrien-f I don't see any new commit.

Reply:
@darccio @adrien-f the reason sarama has this group ID option is because the consumer group is not available otherwise (the usage of consumer groups is through a different api that we currently don't support, see: https://github.com/DataDog/dd-trace-go/issues/2133).

In general, the implementation that we should be looking for reference is the confluent-kafka one: https://github.com/DataDog/dd-trace-go/blob/main/contrib/confluentinc/confluent-kafka-go/kafka.v2/option.go#L114-L116 -> the groupID is fetched from the kafka config actually used by the consumer.